community/bolt/src/main/java/org/neo4j/bolt/v1/runtime/internal/SessionStateMachine.java

```diff
@@ -889,7 +939,11 @@ private void before( Object attachment, Callback cb )
             cb.started( attachment );
         }
 
-        if( interruptCounter.get() > 0 )
+        if ( terminationMark.get() )
```

Comment:
Even though that is true in most cases, this is sitting in a central place in the a protocol stack, so I think it is warranted to be a bit cautious. You could imagine a current or future high-throughput scenario with many very simple client requests where it could have an impact. But if you follow the suggestion above to use an `AtomicReference` I am happy with not nesting :)